server/routes/country_ambassador.js

```diff
@@ -0,0 +1,448 @@
+var express = require('express');
+var passport = require("passport");
+var router = express.Router();
+var schoolSchema = require('../models/schoolSchema');
+var alumniSchema = require('../models/alumniSchema');
+var studentSchema = require('../models/studentSchema');
+var userSchema = require('../models/userSchema');
+var pollSchema = require('../models/polls/pollSchema');
+var pollOptionSchema = require('../models/polls/pollOptionSchema');
+const { sendPollAlert, sendApprovalAlert } = require('./helpers/emailHelpers');
+require('mongoose').Promise = global.Promise
+
+async function isCountryAmbassador(id, country) {
+    let alumni = await alumniSchema.findById(id).populate('user')
+    return (alumni && alumni.user.role.includes('COUNTRY_AMBASSADOR') && alumni.country === country)
+}
+
+/*
+    Return all alumni with the school and access context information populated for given country
+    The country is determined from location of school and NOT where the alumni
+    currently reside
+    @param country
+    @return array of alumni objects
+*/
+async function fetchAllAlumniWithAccessContexts(country) {
+    let schoolsInCountry = await schoolSchema.find({country: country})
+    let alumniData = await alumniSchema.find().where('school').in(schoolsInCountry).populate('school').exec()
+    let alumni = []
+    for (let alumnusModel of alumniData) {
+        let alumnus = alumnusModel.toObject()
+        let userRecordWithAccessContext = await userSchema.findById(alumnusModel.user, {accessContexts: 1})
+        // do not return alumni who do not have an associated user record
+        if (userRecordWithAccessContext) {            
+            alumnus.accessContexts = userRecordWithAccessContext.accessContexts
+            alumni.push(alumnus)
+        }
+    }
+    return alumni
+}
+
+/*
+    Return all students with the school and access context information populated
+    The country is determined from location of school
+    @param country
+    @return array of student objects
+*/
+async function fetchAllStudentsWithAccessContexts(country) {
```

Comment:
This and the above function can be unified, the only difference I can see is whether you use the studentSchema or the alumniSchema

Reply:
noted, makes sense!

Reply:
addressed, thanks for the suggestion!